### PR TITLE
Added integration tests for Pytorch implementation of the ELECTRA model

### LINKED ISSUE #9949: [Good first issue] ELECTRA PyTorch Integration tests
The PyTorch implementation of the ELECTRA model currently has no integration tests. This is problematic as the behavior can diverge without being noticed.

The [test_modeling_electra.py](https://github.com/huggingface/transformers/blob/master/tests/test_modeling_electra.py) file should be updated to include integration testing.

An example of a good modeling integration test is visible in the [test_modeling_bert.py#L552-L565](https://github.com/huggingface/transformers/blob/master/tests/test_modeling_bert.py#L552-L565) file:

https://github.com/huggingface/transformers/blob/1809de5165804666ba6c6a02a9d177f6683869cc/tests/test_modeling_bert.py#L552-L565

Some additional tips:
- The test must be marked as slow using the `@slow` decorator, so as to be run *daily*, and not on every commit of every branch/pull request of this repository.
- The test must be decorated with the `@require_torch` decorator so as to only be run in environments using PyTorch.
- A single test is necessary. If you feel like implementing multiple of these, then sharing the same checkpoint would be ideal so as to reduce download time.
- The TensorFlow implementation already has an integration test, which is visible here:

https://github.com/huggingface/transformers/blob/1809de5165804666ba6c6a02a9d177f6683869cc/tests/test_modeling_tf_electra.py#L253-L267

This test can be translated to PyTorch.

### PR DESCRIPTION
**Pull request by spatil6**
Added integration tests for Pytorch implementation of the ELECTRA model

Fixes #9949 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

@LysandreJik

### CODE REVIEW

**Comment by spatil6** [comment]
@LysandreJik  Need your help here. Not sure, why test cases are failing.

File: tests/test_modeling_electra.py [hunk 1]
```diff
@@ -344,3 +344,19 @@ def test_model_from_pretrained(self):
...
+
+
+@require_torch
+class ElectraModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head_absolute_embedding(self):
+        model = ElectraModel.from_pretrained("google/electra-small-discriminator")
+        input_ids = torch.tensor([[0, 345, 232, 328, 740, 140, 1695, 69, 6078, 1588, 2]])
+        output = model(input_ids)[0]
+        expected_shape = torch.Size((1, 11, 256))
```
Comment (sadakmed) [inline tests/test_modeling_electra.py:356]: the same is here, function used in pytorch is `size()`

File: tests/test_modeling_flaubert.py [hunk 2]
```diff
@@ -399,3 +399,19 @@ def test_model_from_pretrained(self):
         for model_name in FLAUBERT_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
             model = FlaubertModel.from_pretrained(model_name)
             self.assertIsNotNone(model)
+
```
Comment (sadakmed) [inline tests/test_modeling_flaubert.py:402]: this happens when you create a new branch from a development branch, not from master.

**Review by sadakmed — CHANGES_REQUESTED** [review]

File: tests/test_modeling_electra.py (hunk 1 above)
Comment (LysandreJik) [reply tests/test_modeling_electra.py:356]: Same here, `.shape` is an alias to `.size()`

File: tests/test_modeling_electra.py [hunk 3]
```diff
@@ -344,3 +344,19 @@ def test_model_from_pretrained(self):
         for model_name in ELECTRA_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
             model = ElectraModel.from_pretrained(model_name)
             self.assertIsNotNone(model)
+
+
+@require_torch
+class ElectraModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head_absolute_embedding(self):
+        model = ElectraModel.from_pretrained("google/electra-small-discriminator")
```
Comment (LysandreJik) [inline tests/test_modeling_electra.py:353]: Could you use a `ElectraForPreTraining` instead? This model can load the discriminator and has a custom head so it would be nice to test it at well.

Comment (spatil6) [reply tests/test_modeling_electra.py:353]: I have tried for ElectraForPreTraining, however hidden state is coming none for it.

Below is code and output.

model = ElectraForPreTraining.from_pretrained("google/electra-small-discriminator")
input_ids = torch.tensor([[0, 345, 232, 328, 740, 140, 1695, 69, 6078, 1588, 2]])
output = model(input_ids)

output 

ElectraForPreTrainingOutput(loss=None, logits=tensor([[-8.3982, -4.4508, -4.5948, -4.6049, -4.5626, -4.5957, -4.0569, -4.7136,-3.4094, -3.1778, -4.8272]], grad_fn=<SqueezeBackward1>), hidden_states=None, attentions=None)

Comment (LysandreJik) [reply tests/test_modeling_electra.py:353]: Yes, that is the case for `ElectraModel` as well. You didn't put `output_attentions=True`, so it's not outputting the attentions, only the logits (which is what you're evaluating here).

Comment (LysandreJik) [reply tests/test_modeling_electra.py:353]: As pointed out by @spatil6, I meant `output_hidden_states` not being set to `True` and therefore not outputting the `hidden_states` value, not the `attentions`.

**Comment by spatil6** [comment]
closing this PR, due to git conflict.
